lib/matplotlib/dates.py

```diff
@@ -1857,6 +1863,25 @@ def axisinfo(self, unit, axis):
                               default_limits=(datemin, datemax))
 
 
-units.registry[np.datetime64] = DateConverter()
-units.registry[datetime.date] = DateConverter()
-units.registry[datetime.datetime] = DateConverter()
+
+def register_converters():
```

Comment:
This logic should really be in a `_validate_date_converter` validator, no?  otherwise that means that setting the rcParam from anywhere but the matplotlibrc would have no effect.
(And yes, that means we may either need to try to import matplotlib.dates in rcsetup, or, if that's not possible, special case initial import time and re-validate once we import matplotlib.dates).

Reply:
Does a validator get called every time the `rcParam` dict is updated? I guess that would be a good way to do that, if a little magical.

Reply:
Yes

Reply:
Is there an easy way to tell if the validation is the initial one or a subsequent one?  Because if not, I'm not convinced this is worth the bother.